lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1055,12 +1055,19 @@ sub update_backend {
 
 sub insert_module {
     my ($self, $tm) = @_;
-    my $r = $self->modules->find_or_new({name => $tm->{name}});
-    if (!$r->in_storage) {
-        $r->category($tm->{category});
-        $r->script($tm->{script});
+
+    my $r;
+    my $modules = $self->modules;
+    try {
+        $r = $modules->new({name => $tm->{name}, category => $tm->{category}, script => $tm->{script}});
```

Comment:
isn't new+insert = create?

Reply:
It should be, but there are overloads in the `JobModules` class, so i figured it's safer to stay close to the old invocation.

Reply:
Looked at the code quickly, and `create` is a `shift->new_result(shift)->insert` call, so probably ok with our overloads.